reviews/models/mixins.py

```diff
@@ -171,9 +175,33 @@ def save_changes(self, ev):
         now = self.action.date_created if self.action is not None else timezone.now()
         should_publish = self.reviewable.in_public_reviews_state
         if should_publish and not self.reviewable.is_published:
-            self.reviewable.is_published = True
+            if not (self.reviewable.node.preprint_file and self.reviewable.node.preprint_file.node == self.reviewable.node):
+                raise ValueError('Preprint node is not a valid preprint; cannot publish.')
+            if not self.reviewable.provider:
+                raise ValueError('Preprint provider not specified; cannot publish.')
+            if not self.reviewable.subjects.exists():
+                raise ValueError('Preprint must have at least one subject to be published.')
             self.reviewable.date_published = now
-            # TODO EZID (MOD-70)
+            self.reviewable.node._has_abandoned_preprint = False
+            self.reviewable.is_published = True
+            user = ev.kwargs.get('user')
+            auth = Auth(user)
+            self.reviewable.node.add_log(
+                action=NodeLog.PREPRINT_INITIATED,
+                params={
+                    'preprint': self.reviewable._id
+                },
+                auth=auth,
```

Comment:
Creating a node log here is tricky... In pre-moderation, the user here will be the moderator (which should maybe be kept private, depending on `reviews_comments_anonymous`), and calling their action "preprint initiated" seems wrong.

I guess if we're doing node logs, we should probably add actions `PREPRINT_SUBMITTED`, `PREPRINT_PUBLISHED`, and maybe `PREPRINT_UNPUBLISHED`, and only create the appropriate one in this callback. Only `SUBMITTED` should have a user attached to it, the others should leave auth `None` so we don't have to worry about anonymity settings.